pyirf/scripts/make_DL3.py

```diff
@@ -0,0 +1,338 @@
+"""Script to produce DL3 data from DL2 data and a configuration file.
+
+Is it initially thought as a clean start based on old code for reproducing
+EventDisplay DL3 data based on the latest release of the GADF format.
+
+Todo:
+- make some config arguments also CLI ones like in ctapipe-stage1-process
+
+
+"""
+
+# =========================================================================
+#                            MODULE IMPORTS
+# =========================================================================
+
+# PYTHON STANDARD LIBRARY
+
+import argparse
+import os
+
+# THIRD-PARTY MODULES
+
+import numpy as np
+import astropy.units as u
+from astropy.table import Table
+from astropy.coordinates.angle_utilities import angular_separation
+from gammapy.spectrum import cosmic_ray_flux, CrabSpectrum  # UPDATE TO LATEST
+
+# THIS PACKAGE
+
+from pyirf.io.io import load_config, read_FITS
+from pyirf.perf import (
+    CutsOptimisation,
+    CutsDiagnostic,
+    CutsApplicator,
+    IrfMaker,
+    SensitivityMaker,
+)
+
+
+def main():
+
+    # =========================================================================
+    #                   READ INPUT FROM CLI AND CONFIGURATION FILE
+    # =========================================================================
+
+    # INPUT FROM CLI
+
+    parser = argparse.ArgumentParser(description="Produce DL3 data from DL2.")
+
+    parser.add_argument(
+        "--config_file",
+        type=str,
+        required=True,
+        help="A configuration file like pyirf/resources/performance.yaml .",
+    )
+
+    parser.add_argument(
+        "--obs_time",
+        type=str,
+        required=True,
+        help="An observation time written as (value.unit), e.g. '50.h'",
+    )
+
+    parser.add_argument(
+        "--pipeline",
+        type=str,
+        required=True,
+        help="Name of the pipeline that has produced the DL2 files.",
+    )
+
+    parser.add_argument(
+        "--debug", action="store_true", help="Print debugging information."
+    )
+
+    args = parser.parse_args()
+
+    # INPUT FROM THE CONFIGURATION FILE
+
+    cfg = load_config(args.config_file)
+
+    # Add obs. time to the configuration file
+    str_obs_time = args.obs_time.split(".")
+    cfg["analysis"]["obs_time"] = {
+        "value": float(str_obs_time[0]),
```

Comment:
then this becomes `obs_time.value` and `obs_time.unit.to_string('fits')`

Reply:
DONE